core/arch/arch.c

```diff
@@ -3444,8 +3444,10 @@ dr_mcontext_to_priv_mcontext(priv_mcontext_t *dst, dr_mcontext_t *src)
         if (TEST(DR_MC_CONTROL, src->flags)) {
             /* XXX i#2710: mc->lr should be under DR_MC_CONTROL */
             dst->xsp = src->xsp;
+#if !defined(RISCV64)
             if (src->size > offsetof(dr_mcontext_t, xflags))
                 dst->xflags = src->xflags;
+#endif
```

Comment:
Hmm, shouldn't riscv fcsr, and aarch64 fpcr and fpsr, be here under DR_MC_CONTROL (and the docs updated)?  Or do we consider them _MULTIMEDIA?  Or are they only accessible via _ALL -- but it looks like _ALL is not supposed to have anything beyond the other flags.  So we have to pick one, and mention it in the flags docs.  If you don't want to deal w/ aarch64 float regs here (presumably there is no test involving them and we'd want to add one so it is some work) maybe file an issue on it?

Reply:
DR_MC_CONTROL makes sense. I didn't want to disturb Aarch64 and since it wasn't saving FP control registers, I didn't add mine. I'll file an issue.

Reply:
I've added saving of `fcsr`. Haven't touched on Aarch64 logic and opened an issue on that instead.

Reply:
Could we add a `XXX i#nnn` comment on the missing AArch64 fields here pointing at that new issue?

Reply:
Done.